### ISSUE TITLE
Sourcecode Availability

### ISSUE THREAD

**Issue by gked** [pocketnetteam/documentation#6]
On various occasions, it became apparent that Github can take down any code from its servers, and even block its users from accessing their repos.  I wanted to propose that the PocketNet team moves its repo to Gitlab or Gitea. Understandably, it is less convenient for those making frequent changes in the repo and who have to use Github for other projects. In addition, there is an additional maintenance costs involved in maintaining the server hosting the GitLab instance. I do believe, however that is a small price to pay should this repo get disrupted. In addition, there is a anonymity benefit to contributors who are not necessarily going to be using the same userIds on GitLab repo and thus breaking relationship link from their other github activity.

**Comment by Dennis-Volk** [comment]
What do you think about Radicle? This has been proposed before. Another alternative is to store code in Pocketnet video servers with a hash in the blockchain along with digital signatures. This is a valuable discussion we are having.

**Comment by andyoknen** [comment]
I'm also leaning more towards decentralized solutions, instead of centralized current services or personal servers.

Radical looks nice, I'll study what they offer and we can discuss the reality of switching to them

**Comment by gked** [comment]
I think as long as the goals are achieved either solution could work:

Here are a few goals/criteria that came to mind:

- Minimize the influence of the central authority, i.e. Microsoft.
- Similar technology - git 
- Similar capability as GitHub provides (build systems, web interface for managing tickets, etc)
- Simplicity of porting the repo
- Preserve repo history (commit history, tickets, builds, etc)
- Minimal time cost to maintain the solution 
- Minimal financial cost to maintain the solution (I am not too familiar with how well this project is funded but putting it down just in case)
- Better anonymity

What I like about being able to host using Bastyon blockchain is the fact that it is a testament to the versatility of the technology and would be another successful(if worked) use case for the use of Bastyon.

**Comment by andyoknen** [comment]
I agree! As for the placement in the blockchain, so far I see only the need to save a minimal set of information (release versions and file hashes).

I think there are many projects like Radical with decentralized git - the question is how popular they are to use them as a working platform..

**Comment by gked** [comment]
If you end up leaning towards a non-blockchain-based solution, here is a list of existing open-source project and code management projects that just need a server instance spun up and to be installed- https://github.com/awesome-selfhosted/awesome-selfhosted#software-development---project-management
